### ISSUE TITLE
Suggestion: easily open up endless integration options while reducing code and complexity

### ISSUE THREAD

**Issue by JayGuerette** [tphakala/birdnet-go#162]
Instead of adding/maintaining new databases, notification options, API integrations, & species configurations, just add a hooks that are called every time there is a realtime detection.

Define in the config the name of the executable including argument templates like these examples:

- curl 'https://myserver/detection?name={{common_name}}&score={{confidence}}'
- redis-cli -u redis://myserver:6379/0 HINCRBY birdnet {{common_name:/\s/_/}} 1
- notify-send -a BirdNet 'New detection: {{scientific_name}}'
- psql -U birdnet -h myserver -p 25060 -d mydb -c "INSERT INTO mytable VALUES ('{{date}}', '{{time}}', '{{common_name}}'); "

In addition to arguments expose the complete note record as environment variables:
BIRDNET_DATE
BIRDNET_TIME
BIRDNET_COMMON_NAME
BIRDNET_CONFIDENCE
etc.

Based on this, users could use shell scripts instead for more complex integrations:
```
#! /bin/bash
count=$(valkey-cli --raw -u redis://myserver:6379/0 HINCRBY birdnet ${BIRDNET_COMMON_NAME// /_} 1)
[[ $count -eq 1 ]] && notify-send -a BirdNet 'Previously undetected bird' "$BIRDNET_COMMON_NAME"
```

Any allowed/configured executable must by placed in or symlinked from a preconfigured directory. Then you expose it to your container: docker run -v $HOME/hooks:/hooks ghcr.io/tphakala/birdnet-go:latest

In the examples above we have Postgresql, notifications, redis/valkey, and custom REST API support with no additional code base dependencies or complexity.

The currently built-in birdweather integration could be extracted from the code base and put into a stand-alone executable that is called as a hook.

**Comment by tphakala** [comment]
Yeah it is quite easy to do, and I don't see any reason why not to implement this. Hoever I would not like to exclud any direct integrations out either as while this allows any integration you ever need, it requires quite complex setups from end user.

**Comment by tphakala** [comment]
Closing as resolved. The push notification system now provides exactly this capability through:
- **Script provider**: Execute custom scripts on detection events with JSON/ENV input (#882 Phase 3)
- **Webhook provider**: Send HTTP requests to any endpoint on detection (#882 Phase 4, PR #1352)
- **Custom actions**: Species-specific script execution (with error surfacing via PR #1761)

These features allow arbitrary integrations triggered by detections, which was the core request.
